itests/kit/node_unmanaged.go

```diff
@@ -24,3 +60,730 @@ type TestUnmanagedMiner struct {
+	}
+	tm.Libp2p.PeerID = peerId
+	tm.Libp2p.PrivKey = privkey
+
+	return &tm
+}
+
+func (tm *TestUnmanagedMiner) AssertNoPower(ctx context.Context) {
+	p := tm.CurrentPower(ctx)
+	tm.t.Logf("MinerB RBP: %v, QaP: %v", p.MinerPower.QualityAdjPower.String(), p.MinerPower.RawBytePower.String())
+	require.True(tm.t, p.MinerPower.RawBytePower.IsZero())
+}
+
+func (tm *TestUnmanagedMiner) CurrentPower(ctx context.Context) *api.MinerPower {
+	head, err := tm.FullNode.ChainHead(ctx)
+	require.NoError(tm.t, err)
+
+	p, err := tm.FullNode.StateMinerPower(ctx, tm.ActorAddr, head.Key())
+	require.NoError(tm.t, err)
+
+	return p
+}
+
+func (tm *TestUnmanagedMiner) AssertPower(ctx context.Context, raw uint64, qa uint64) {
+	req := require.New(tm.t)
+	p := tm.CurrentPower(ctx)
+	tm.t.Logf("MinerB RBP: %v, QaP: %v", p.MinerPower.QualityAdjPower.String(), p.MinerPower.RawBytePower.String())
+	req.Equal(raw, p.MinerPower.RawBytePower.Uint64())
+	req.Equal(qa, p.MinerPower.QualityAdjPower.Uint64())
+}
+
+func (tm *TestUnmanagedMiner) mkAndSavePiecesToOnboard(_ context.Context, sectorNumber abi.SectorNumber, pt abi.RegisteredSealProof) []abi.PieceInfo {
+	paddedPieceSize := abi.PaddedPieceSize(sectorSize)
+	unpaddedPieceSize := paddedPieceSize.Unpadded()
+
+	// Generate random bytes for the piece
+	randomBytes := make([]byte, unpaddedPieceSize)
+	_, err := io.ReadFull(rand.Reader, randomBytes)
+	require.NoError(tm.t, err)
+
+	// Create a temporary file for the first piece
+	pieceFileA := requireTempFile(tm.t, bytes.NewReader(randomBytes), uint64(unpaddedPieceSize))
+
+	// Generate the piece CID from the file
+	pieceCIDA, err := ffi.GeneratePieceCIDFromFile(pt, pieceFileA, unpaddedPieceSize)
+	require.NoError(tm.t, err)
+
+	// Reset file offset to the beginning after CID generation
+	_, err = pieceFileA.Seek(0, io.SeekStart)
+	require.NoError(tm.t, err)
+
+	unsealedSectorFile := requireTempFile(tm.t, bytes.NewReader([]byte{}), 0)
+	defer func() {
+		_ = unsealedSectorFile.Close()
+	}()
+
+	// Write the piece to the staged sector file without alignment
+	writtenBytes, pieceCID, err := ffi.WriteWithoutAlignment(pt, pieceFileA, unpaddedPieceSize, unsealedSectorFile)
+	require.NoError(tm.t, err)
+	require.EqualValues(tm.t, unpaddedPieceSize, writtenBytes)
+	require.True(tm.t, pieceCID.Equals(pieceCIDA))
+
+	// Create a struct for the piece info
+	publicPieces := []abi.PieceInfo{{
+		Size:     paddedPieceSize,
+		PieceCID: pieceCIDA,
+	}}
+
+	// Create a temporary file for the sealed sector
+	sealedSectorFile := requireTempFile(tm.t, bytes.NewReader([]byte{}), 0)
+	defer func() {
+		_ = sealedSectorFile.Close()
+	}()
+
+	// Update paths for the sector
+	tm.sealedSectorPaths[sectorNumber] = sealedSectorFile.Name()
+	tm.unsealedSectorPaths[sectorNumber] = unsealedSectorFile.Name()
+	tm.cacheDirPaths[sectorNumber] = filepath.Join(tm.cacheDir, fmt.Sprintf("%d", sectorNumber))
+
+	// Ensure the cache directory exists
+	_ = os.Mkdir(tm.cacheDirPaths[sectorNumber], 0755)
```

Comment:
I think this isn't even necessary, I've been running proofs manually and it seems to make the cache itself, perhaps try removing this and see if it passes

Reply:
@rvagg The tests fail for me without this. I'm surprised it works for you. Are you sure you don't have a `mkDir` anywhere ? 

```
    node_unmanaged.go:636: Miner f01003: Running proof type 5 SealPreCommitPhase1 for sector 101...
    node_unmanaged.go:652: 
        	Error Trace:	/Users/aarshshah/src/filecoin/lotus/itests/kit/node_unmanaged.go:652
        	            				/Users/aarshshah/src/filecoin/lotus/itests/kit/node_unmanaged.go:240
        	            				/Users/aarshshah/src/filecoin/lotus/itests/manual_onboarding_test.go:77
        	Error:      	Received unexpected error:
        	            	No such file or directory (os error 2)
        	Test:       	TestManualCCOnboarding
        	Messages:   	Miner f01003: SealPreCommitPhase1 failed for sector 101
```

Reply:
oh, maybe it does then, I kept a mkdir in my local version but I thought at one stage I saw it doing it itself without needing me to, probably mistaken